src/coreclr/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -1871,6 +1871,107 @@ void CodeGen::genBaseIntrinsic(GenTreeHWIntrinsic* node, insOpts instOptions)
             break;
         }
 
+        case NI_Vector128_op_Division:
+        case NI_Vector256_op_Division:
+        {
+            // We can emulate SIMD integer division by converting the 32-bit integer -> 64-bit double,
+            // perform a 64-bit double divide, then convert back to a 32-bit integer. This is generating
+            // something similar to the following managed code:
+            //      if (Vector128.EqualsAny(op2, Vector128<int>.Zero))
+            //      {
+            //          throw new DivideByZeroException();
+            //      }
+            //
+            //      Vector128<int> overflowMask =
+            //          Vector128.Equals(op1, Vector128.Create(int.MaxValue)
```

Comment:
this should be `MinValue`, not `MaxValue`.

it's `MinValue / -1` that overflows, since that produces `2147483648`, which isn't representable.

Reply:
Thanks, fixed.